### PR TITLE
Local records: better layout to avoid text overlapping in smaller screens

### PR DESCRIPTION
**Pull request by rdwebdesign**
### What does this PR aim to accomplish?

Second part of #2854 fix.
Avoid text overlapping in smaller screens.

### How does this PR accomplish the above?*

- Using a single column in medium and small screens. Only use 2 columns in large screens.
- Small adjustments to grid holding the datatables control elements.
- Replacing the collapsible box: moving the input fields to the table footer and notes to the bottom.

image

mobile

### Link documentation PRs if any are needed to support this PR:

none

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

File: settings-dnsrecords.lp [hunk 1]
```diff
@@ -14,125 +14,96 @@ PageTitle = "Local DNS Settings"
...
+                                <tr>
+                                    <th>Domain</th>
+                                    <th>IP</th>
+                                    <th></th>
+                                </tr>
                             </thead>
+                            <tfoot class="add-new-item">
+                                <tr>
+                                    <th>
+                                        <input id="Hdomain" type="url" class="form-control" data-configkeys="hosts" placeholder="Add a domain (example.com or sub.example.com)" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```
Comment (yubiuser) [inline settings-dnsrecords.lp]:
```suggestion
                                        <input id="Hdomain" type="url" class="form-control" data-configkeys="hosts" placeholder="Domain" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```

File: settings-dnsrecords.lp [hunk 2]
```diff
@@ -14,125 +14,96 @@ PageTitle = "Local DNS Settings"
...
+                                    </th>
+                                    <th>
+                                        <input id="Hip" type="text" class="form-control" data-configkeys="hosts" placeholder="Associated IP address" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```
Comment (yubiuser) [inline settings-dnsrecords.lp]:
```suggestion
                                        <input id="Hip" type="text" class="form-control" data-configkeys="hosts" placeholder="Associated IP" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```

File: settings-dnsrecords.lp [hunk 3]
```diff
@@ -14,125 +14,96 @@ PageTitle = "Local DNS Settings"
...
+                                    <th>Domain</th>
+                                    <th>Target</th>
+                                    <th>TTL <i class="small">(optional)</i></th>
+                                    <th></th>
+                                </tr>
                             </thead>
+                            <tfoot class="add-new-item">
+                                <tr>
+                                    <th>
+                                        <input id="Cdomain" type="url" class="form-control" data-configkeys="cnameRecords" placeholder="Add a domain (example.com or sub.example.com)" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```
Comment (yubiuser) [inline settings-dnsrecords.lp]:
```suggestion
                                        <input id="Cdomain" type="url" class="form-control" data-configkeys="cnameRecords" placeholder="Domain" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```

File: settings-dnsrecords.lp [hunk 4]
```diff
@@ -14,125 +14,96 @@ PageTitle = "Local DNS Settings"
...
+                                    </th>
+                                    <th>
+                                        <input id="Ctarget" type="url" class="form-control" data-configkeys="cnameRecords" placeholder="Associated Target Domain" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```
Comment (yubiuser) [inline settings-dnsrecords.lp]:
```suggestion
                                        <input id="Ctarget" type="url" class="form-control" data-configkeys="cnameRecords" placeholder="Target Domain" autocomplete="off" spellcheck="false" autocapitalize="none" autocorrect="off">
```

**Review by yubiuser — CHANGES_REQUESTED** [review]
It's to narrow in boxed layout

Screenshot at 2023-11-22 08-14-40
Screenshot at 2023-11-22 08-14-52

**Comment by rdwebdesign** [comment]
What do you think about this?

image

File: settings-dnsrecords.lp (hunk 1 above)
Comment (rdwebdesign) [reply settings-dnsrecords.lp]: Done

File: settings-dnsrecords.lp (hunk 2 above)
Comment (rdwebdesign) [reply settings-dnsrecords.lp]: Done

File: settings-dnsrecords.lp (hunk 3 above)
Comment (rdwebdesign) [reply settings-dnsrecords.lp]: Done

File: settings-dnsrecords.lp (hunk 4 above)
Comment (rdwebdesign) [reply settings-dnsrecords.lp]: Done

**Comment by rdwebdesign** [comment]
I saw a small intermittent glitch during the tests. 

If I reload the page many times, sometimes the column widths are not correct (see the column with the button):

image

Looking at the browser console, the columns width were set to zero on the first table (the one showing the issue).
I think the table is drawn before the Datatables calculates the column widths.
The widths are fixed if the windows is manually resized. Using `.columns.adjust().draw()` did not fix the issue.

Adding `autoWidth: false` fixed the issue:

image

**Review by yubiuser — APPROVED** [review]

**Pull request merged**